packages/utils/src/storage/db.rs

```diff
@@ -1,384 +1,306 @@
+        TableIter {
+            inner: self.map.iter(),
         }
+    }
+}
 
-        loop {
-            if counter.reached(task_target, op_target) {
-                break;
-            }
+pub struct TableIter<'a, K, V> {
+    inner: dashmap::iter::Iter<'a, K, V>,
+}
 
-            // give it a little time to let other threads run
-            sleep(Duration::from_millis(1));
-        }
-    }
+impl<'a, K, V> Iterator for TableIter<'a, K, V>
+where
+    K: Eq + Hash,
+{
+    type Item = TableEntry<'a, K, V>;
 
-    #[tokio::test]
-    #[ignore]
-    async fn storage_concurrent_in_memory() {
-        storage_concurrent_inner(StorageKind::InMemory, 20, 1000).await;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next().map(TableEntry)
     }
+}
 
-    #[tokio::test]
-    #[ignore]
-    async fn storage_concurrent_on_disk() {
-        let temp_dir = TempDir::new().unwrap();
-        let db_path = temp_dir.path().join("test_db.redb");
-
-        // incresaing the number of ops *dramatically* increases test time
-        storage_concurrent_inner(StorageKind::OnDisk(db_path), 10, 100).await;
+pub struct TableEntry<'a, K, V>(RefMulti<'a, K, V>);
 
-        // just make sure we didn't drop it
-        let _temp_dir = temp_dir;
+impl<'a, K, V> TableEntry<'a, K, V>
+where
+    K: Eq + Hash,
+{
+    pub fn pair(&self) -> (&K, &V) {
+        (self.0.key(), self.0.value())
     }
+}
 
-    async fn storage_concurrent_inner(
-        storage_kind: StorageKind,
-        task_target: usize,
-        op_target: usize,
-    ) {
-        let storage = new_storage(storage_kind);
-
-        // stash something at the beginning so we're guaranteed to get a valid read
-        storage.set(TABLE, 1u32, &1u32).unwrap();
-
-        let counter = Counter::new();
-
-        let mut futures = FuturesUnordered::new();
-
-        for task_num in 0..task_target {
-            for _ in 0..op_target {
-                futures.push({
-                    let storage = storage.clone();
-                    let counter = counter.clone();
-                    async move {
-                        if task_num % 2 == 0 {
-                            spawn_blocking(move || {
-                                storage.set(TABLE, 1u32, &1u32).unwrap();
-                            })
-                            .await
-                            .unwrap();
-                        } else {
-                            let value = spawn_blocking(move || {
-                                storage.get(TABLE, 1).unwrap().unwrap().value()
-                            })
-                            .await
-                            .unwrap();
-
-                            assert_eq!(value, 1u32);
-                        }
-                        counter.increment(task_num);
-                    }
-                });
-            }
-        }
-
-        while (futures.next().await).is_some() {}
+#[cfg(test)]
+mod tests {
```

Comment:
I agree with deleting the old test code which was exploring in-memory vs. on-disk, but we should still have some test showing that our db access works concurrently (eventually we should probably do more fancy things like fuzz-testing, but a basic "it works with async" test is a good baseline)

not a blocker though, since the new implementation pretty much guarantees that's okay at compiletime